### ISSUE TITLE
Remove error output from lapack_full_matrix.cc

### ISSUE THREAD

**Issue by bangerth** [dealii/dealii#15363]
As pointed out by Tao Jin on the mailing list, the file `source/lac/lapack_full_matrix.cc` contains code that prints errors to the console -- for example this:
```
  // Negative return value implies a wrong argument. This should be internal.
  Assert(info >= 0, ExcInternalError());
  if (info != 0)
    std::cerr << "LAPACK error in syevx" << std::endl;
```
This is a nuisance and does not follow out coding convention today (the specific code here dates back to one of the very first revisions of that file, in 2005). 

We ought to remove this and replace it with proper error checking the way we usually do it. This will require looking up what the various error values returned by LAPACK actually mean. Here, for syevx, we can look at https://www.intel.com/content/www/us/en/docs/onemkl/developer-reference-fortran/2023-1/syevx.html which states:
```
info
INTEGER.

If info = 0, the execution is successful.

If info = -i, the i-th parameter had an illegal value.

If info = i, then i eigenvectors failed to converge; their indices are stored in the array ifail.
```
So presumably, this should have been translated into something like
```
  Assert (info >= 0, ExcInternalError());   // negative return values correspond to wrong arguments -> internal error
  if (info > 0)
    throw SomeKindOfError();  // encode the information about what failed to converge, and perhaps more information

  // info == 0; everything is fine
  ...
```

**Comment by taojinllnl** [comment]
Hi Wolfgang,

I would be delighted to try to fix this issue.
Best,

Tao

**Comment by bangerth** [comment]
@taojinllnl Let us know if you have questions or need help!

**Comment by bangerth** [comment]
Fixed by #15377.
